File: js/p4.js
var calc = function() {
    for (var n = 999*999; n >= 100*100; n--) {
        var s = n.toString();
        if (s != s.split('').reverse().join('')) continue;

        for (var i = 999; i >= 99; i--) {
            var d = n / i;
            if (n % i == 0 && d >= 100 && d <= 999) {
                return n;
            }
        }
    }
};

console.log(calc());

File: lua/p4.lua
function calc()
    for n = 999*999, 100*100, -1 do
        s = tostring(n)
        if s == string.reverse(s) then
            for i = 999, 99, -1 do
                d = n / i
                if n % i == 0 and d >= 100 and d <= 999 then
                    return n
                end
            end
        end
    end
end

print(calc())
